tools/Raven.Debug/Dump/Dumper.Windows.cs

```diff
@@ -0,0 +1,104 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Win32.SafeHandles;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Dump
+{
+    public partial class Dumper
+    {
+        private static class Windows
+        {
+            internal static Task CollectDumpAsync(Process process, string outputFile, DumpTypeOption type)
+            {
+                // We can't do this "asynchronously" so just Task.Run it. It shouldn't be "long-running" so this is fairly safe.
+                return Task.Run(() =>
+                {
+                    // Open the file for writing
+                    using (var stream = new FileStream(outputFile, FileMode.Create, FileAccess.ReadWrite, FileShare.None))
+                    {
+                        var exceptionInfo = new NativeMethods.MINIDUMP_EXCEPTION_INFORMATION();
+                        var dumpType = type == DumpTypeOption.Mini ? NativeMethods.MINIDUMP_TYPE.MiniDumpWithThreadInfo :
```

Comment:
I wonder, for both Linux & Windows, if we can skip dumping the mmap files?
We typically don't need or want them